roles/custom/matrix-media-repo/tasks/setup_install.yml

```diff
@@ -77,6 +77,58 @@
       changed_when: true
       when: "matrix_media_repo_git_pull_results.changed | bool or matrix_media_repo_docker_image_check_result.stdout == ''"
 
+- name: Check existence of media-repo signing key
+  ansible.builtin.stat:
+    path: "{{ matrix_media_repo_config_path }}/{{ matrix_media_repo_identifier }}.signing.key"
+  register: matrix_media_repo_signing_key_stat
+
+- when: "matrix_media_repo_generate_signing_key | bool and not (matrix_media_repo_signing_key_stat.stat.exists | bool)"
+  block:
```

Comment:
All the magic happens in this guard block, but.. The `when` will be satisfied early on (in the first sub-task) and there are more tasks after that.

This means that if the playbook dies after the first task creates the file and someone re-runs the playbook later on, we would never again enter this `when` block again. The job would effectively be half-done, without a chance to make it fully done except for figuring out what happened and deleting the `{{ matrix_media_repo_config_path }}/{{ matrix_media_repo_identifier }}.signing.key` file.

-----

A potential solution is:

1. Continue guarding on ` {{ matrix_media_repo_config_path }}/{{ matrix_media_repo_identifier }}.signing.key` as you do now
2. Generate  `{{ matrix_media_repo_config_path }}/{{ matrix_media_repo_identifier }}.signing.key.TEMPORARY` instead of  `{{ matrix_media_repo_config_path }}/{{ matrix_media_repo_identifier }}.signing.key`
3. Work with `{{ matrix_media_repo_config_path }}/{{ matrix_media_repo_identifier }}.signing.key.TEMPORARY` in the `block`
4. Move `{{ matrix_media_repo_config_path }}/{{ matrix_media_repo_identifier }}.signing.key.TEMPORARY` to `{{ matrix_media_repo_config_path }}/{{ matrix_media_repo_identifier }}.signing.key` as a last step in this block